src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/SunPKCS11.java

```diff
@@ -1281,10 +1292,9 @@ private void initToken(CK_SLOT_INFO slotInfo) throws PKCS11Exception {
         // the earliest entry.  When asked for "DES/CBC/PKCS5Padding", we
         // return a CKM_DES_CBC_PAD.
         final Map<Descriptor,Integer> supportedAlgs =
-                                        new HashMap<Descriptor,Integer>();
+                new HashMap<Descriptor,Integer>();
 
-        for (int i = 0; i < supportedMechanisms.length; i++) {
-            long longMech = supportedMechanisms[i];
+        for (long longMech : supportedMechanisms) {
```

Comment:
Is the code under `if (!config.isEnabled(longMech)) {` still needed? Looks to me that we will be iterating over enabled mechanisms now.

Reply:
Just for the record, I want to note that we will not longer be showing information about supported but disabled mechanisms here.

Reply:
Right, I agree that it's better to show info about supported but disabled mechanisms.